docs/architecture/metric-tracking.md

```diff
@@ -0,0 +1,612 @@
+# Metric Tracking System - Architecture
```

Comment:
Suggest this document would be better housed in docs/features rather than docs/architecture.